front/ticket.form.php

```diff
@@ -76,7 +76,9 @@
     }
     Html::back();
 } else if (isset($_POST['update'])) {
-    $track->check($_POST['id'], UPDATE);
+    if (!$track->canAssign()) {
```

Comment:
I'm not sure about adding a specifity in the controller part.
I would prefer a change in canUpdatexxx methods i think.

More globally, this is very strange that 9.5.x is impacted as assign right is already tested as i remember (maybe not as you complete the test method).
Where is more strange, there is a case for this in Ticket::canUpdate 
![image](https://user-images.githubusercontent.com/418844/157239484-d9baf647-cdaa-4f12-838e-0f83a0173774.png)

Reply:
maybe it's just adding a line `!$this->canAssign()` here : https://github.com/glpi-project/glpi/blob/master/src%2FTicket.php#L539

Reply:
I made those changes and there are 3 test failures currently (only checked Ticket tests). I'll apply the improved test error messages from #10824 and look into why it fails some more.

Reply:
There is one test that fails with this method that makes it look like having the right to assign also gives global update permission for tickets. The access denied screen when trying to just assign users has returned.